src/read.c

```diff
@@ -1248,15 +1274,13 @@ void ReadFuncExpr (
         Match( S_SEMICOLON, ";", STATBEGIN|S_END|follow );
     }
 
-    /* 'function( ... arg )' takes a variable number of arguments              */
-    if (narg >= 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) ) )
+    /* 'function( a,b, ...p )' takes a variable number of arguments           */
+    /* Also, we special case function(arg)                                   */
+    if (isvarg || ( narg == 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) )) )
       {
-	/* TODO: remove this before the next non-beta release */
-	if (narg > 1)
-	  SyntaxWarning("New syntax used -- intentional?");
 	narg = -narg;
       }
-	
+      
```

Comment:
If you change whitespace here, perhaps just remove them (and leave an empty line)?